File: bindings/java/com/microsoft/cognitiveservices/speech/SpeechRecognitionResult.java
package com.microsoft.cognitiveservices.speech;
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//

import com.microsoft.cognitiveservices.speech.RecognitionStatus;
import com.microsoft.cognitiveservices.speech.internal.RecognitionResult;
import com.microsoft.cognitiveservices.speech.internal.ResultPropertyValueCollection;
import com.microsoft.cognitiveservices.speech.internal.ResultProperty;

/**
  * Defines result of speech recognition.
  */
public class SpeechRecognitionResult {

    private String resultId;
    private RecognitionStatus reason;
    private String recognizedText;
    private ResultPropertyValueCollection properties;

    protected SpeechRecognitionResult(RecognitionResult result) {
        resultId = result.getResultId();
        recognizedText = result.getText();
        reason = RecognitionStatus.values()[result.getReason().swigValue()];
        properties = result.getProperties();
    }

    /**
      * Specifies the result identifier.
      * @return Specifies the result identifier.
      */
    public String getResultId() {
        return resultId;
    }
    
    /**
      * Specifies status of the result.
      * @return Specifies status of the result.
      */
    public RecognitionStatus getReason() {
        return reason;
    }
    

    /**
      * Presents the recognized text in the result.
      * @return Presents the recognized text in the result.
      */
    public String getRecognizedText() {
        return recognizedText;
    }

    /**
    * In case of an unsuccessful recognition, provides a brief description of an occurred error.
    * This field is only filled-out if the recognition status (@see RecognitionStatus) is set to Canceled.
    * @return a brief description of an error.
    */
    public String getErrorDetails() {
        return properties.get(ResultProperty.ErrorDetails).getString();
    }

    /**
    * A string containing Json serialized recognition result as it was received from the service.
    * @return Json serialized representation of the result.
    */
    public String getJson() {
      return properties.get(ResultProperty.Json).getString();
    }
    
    /**
      * Returns a String that represents the speech recognition result.
      * @return A String that represents the speech recognition result.
      */
    @Override
    public String toString() {
        return "ResultId:" + resultId + " Status:" + reason + " Recognized text:<" + recognizedText + ">.";
    }
}

File: source/bindings/js/src/sdk/speech.browser/SpeechRecognitionResult.ts
//
// copyright (c) Microsoft. All rights reserved.
// licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
import { ISpeechProperties, RecognitionStatus } from "./Exports";

/**
 * Defines result of speech recognition.
 */
export class SpeechRecognitionResult {
    /**
     * Specifies the result identifier.
     * @return Specifies the result identifier.
     */
    public resultId: string;

    /**
     * Specifies status of the result.
     * @return Specifies status of the result.
     */
    public reason: RecognitionStatus;

    /**
     * Presents the recognized text in the result.
     * @return Presents the recognized text in the result.
     */
    public text: string;

    /**
     * Duration of recognized speech in milliseconds.
     * @return Duration of recognized speech in milliseconds.
     */
    public duration: number;

    /**
     * Offset of recognized speech in milliseconds.
     * @return Offset of recognized speech in milliseconds.
     */
    public offset: number;

    /**
     * In case of an unsuccessful recognition, provides a brief description of an occurred error.
     * This field is only filled-out if the recognition status (@see RecognitionStatus) is set to Canceled.
     * @return a brief description of an error.
     */
    public errorDetails: string;

    /**
     * A string containing Json serialized recognition result as it was received from the service.
     * @return Json serialized representation of the result.
     */
    public json: string;

    /**
     *  The set of properties exposed in the result.
     * @return The set of properties exposed in the result.
     */
    public properties: ISpeechProperties;
}
